spec/factories/leagues.rb

```diff
@@ -4,6 +4,8 @@
     gameday_duration 7
 
     factory :league_with_teams, class: League, parent: :event_with_teams, traits: [:has_dates] do
+      game_mode :round_robin
```

Comment:
Isn't it better to set the `game_mode` randomly here and specify it in the respective tests?

Reply:
The value is defaulted to :round_robin right now because it is the only working game mode atm I think. If we change this we'd have to change it in the default factory aswell, which would cause a bigger changing of tests.
I want to start a test refactoring anyways since a lot of the older event tests seem a little of, so I'd add that to the ToDo list on that issue.

Also having subfactories for "league_with_teams" doesn't really make any sense which is why I added the trait. I don't want to remove this right here either though, because it would mean that I have to change another 10 or so files. And the PR is big enough as it is ^^"